### PR TITLE
Fix ArrayView and numpy subclassing

### PR DESCRIPTION
**Pull request by michalk8**
Hi,
I have a `numpy.ndarray` subclass with some column annotations (also `ndarrays`), including a
view-like object. Problem is that taking view of `adata` object first converts this to `np.ndarray`,
effectively removing my subclass (`arr = np.asarray(input_array)`). This PR fixes this such that only non-arrays are converted to arrays, rest is left the same.

### CODE REVIEW

File: anndata/_core/views.py [hunk 1]
```diff
@@ -69,7 +69,9 @@ def __new__(
         input_array: Sequence[Any],
         view_args: Tuple["anndata.AnnData", str, Tuple[str, ...]] = None,
     ):
-        arr = np.asarray(input_array).view(cls)
+        if not isinstance(input_array, np.ndarray):
+            input_array = np.array(input_array)
+        arr = input_array.view(cls)
```
Comment (ivirshup) [inline anndata/_core/views.py]:
```suggestion
        arr = np.asanyarray(input_array).view(cls)
```

Could this work instead? I think it should if your class is literally a np.ndarray subclass.

Comment (michalk8) [reply anndata/_core/views.py]: Thanks for the suggestion! I've added a test and it works.

**Comment by michalk8** [comment]
Any news on this? I wanted to fix the code-quality check (just for the test - either get overriding builtin when using `type` as an argument or parameters differ...).
@ivirshup would be great if you think everything is on point and then merge this.

**Comment by ivirshup** [comment]
Sorry, just been busy! I think this looks good, thanks!

**Pull request merged**